#919: Require user to agree to license

hohwille opened the pull request:
fixes #919

jan-vcapgemini reviewed (commented):
LGTM. I've added some CRs and questions, please resolve.

hohwille commented on cli/src/main/package/setup.bat:
```diff
@@ -72,6 +72,7 @@ echo "%BASH%" -l -c "cd \"%CD%\";./setup"
 "%BASH%" -l -c "cd \"%CD%\";./setup"
 if %ERRORLEVEL% neq 0 (
   echo %_fBRed%Error occurred while running setup of IDEasy in bash.%_RESET%
+  pause
```
See commit comment: https://github.com/devonfw/IDEasy/pull/948/commits/b874c66df3b003c0d067e30702d8b1bd8b30ed0e

hohwille commented on cli/src/main/package/setup:
```diff
@@ -30,13 +25,19 @@ function doSetupInConfigFile() {
 fi
 echo "Setting up IDEasy in ${PWD}"
 cd ..
+if [ "${PWD/*\//}" != "projects" ]; then
+  echo -e "\033[93mInvalid IDE_ROOT path $PWD - you need to rename '${PWD/*\//}' to 'projects'.\033[39m" >&2
+  exit 1
```
Because some users use 7-zip `extract here...` feature and get a folder named like e.g. `ide-cli-2024.09.002-beta-windows-x64.tar` and this will turn out to be their `IDE_ROOT` folder. After they have setup their project they do not dare again to change this folder since they are unable to make it work to change `IDE_ROOT` consistently and end up forever with a messy setup.
I have seen so many users doing crazy or stupid things most probably by accident but ending up working this way for years. It is better to enforce standards and align environments a little more strict than too lax allowing every nonsense.
BTW: I first also wanted to accept alternative folder names like e.g. `Projekte` and even implemented case insensitive comparison in Bash but then I asked my self what about Polish, Indian, or other translations of `projects` and went for KISS.

hohwille commented:
Resolved the review comments.
Additionally I also addressed the main problem/concern that the current implementation could cause the situation that the user gets totally blocked and unable to open any bash terminal any more. This is now also fixed and therefore ready for merge.
